### PR TITLE
Existing envelope

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-719

### Change description ###
Code improvement - made result of checking envelope existence Optional rather than using null and some brushing up.

Also noticed that envelope existence runs 2 times - before acquiring lease and after acquiring lease. I assume this is justified but whoever will be reviewing this PR please confirm this workflow is correct.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by rkondratowicz** [comment]
I don't see this change as an improvement.
There's more code now and then new method `tryLeaseAndProcessZipFile` that wraps `leaseAndProcessZipFile` adds confusion imho.

**Comment by apereverzin** [comment]
Using Optional instead of checking for null and introducing a method instead of _else - if - else_ - this makes logic cleaner imho.

Please, also reply to my question in the description

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java [hunk 1]
```diff
@@ -69,8 +69,8 @@ void processBlobs_should_not_call_envelopeProcessor_if_failed_to_load_file() thr
         given(blob.getUri()).willReturn(URI.create("file.zip"));
         given(container.getBlockBlobReference("file.zip")).willReturn(cloudBlockBlob);
         given(container.getName()).willReturn("cont");
-        given(envelopeProcessor.getEnvelopeByFileAndContainer("cont", "file.zip"))
-            .willReturn(null);
+        given(envelopeProcessor.getEnvelopeIdByFileAndContainer("cont", "file.zip"))
+            .willReturn(Optional.empty());
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java:73]: do we need a test with non empty optional too?

Comment (apereverzin) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java:73]: Not right now. BlobProcessorTaskTest always contained this only method effectively making BlobProcessorTask not unit tested (all code coverage was provided only by integration tests). After refactoring BlobProcessorTask will become unit testable and more methods will be added to BlobProcessorTaskTest.

**Comment by apereverzin** [comment]
Not necessary
